model_compression_toolkit/core/pytorch/pytorch_implementation.py

```diff
@@ -474,3 +516,66 @@ def get_node_mac_operations(self,
             return kernel_shape[0] * kernel_shape[1]
         else:
             return 0
+
+    def apply_second_moment_correction(self, quantized_model, core_config, representative_data_gen, graph):
+        """
+        Apply second moment statistics correction to graph.
+
+        Args:
+            quantized_model: Framework's model to apply second moment correction on.
+            core_config: QuantizationConfig of how the model should be quantized.
+            representative_data_gen: Dataset to use for retrieving images for the models inputs.
+            graph: Graph to update the parameters after the second moment correction.
+
+        Returns:
+            A function that applies second moment correction.
+        """
+        model = copy.deepcopy(quantized_model)
+        set_model(model)
+
+        # Move every BN to train mode
+        for module in model.modules():
+            if isinstance(module, torch.nn.BatchNorm2d):
+                module.train()
+
+        for _ in tqdm(range(core_config.quantization_config.weights_second_moment_iters)):
+            with torch.no_grad():
+                model(*to_torch_tensor(representative_data_gen()))
+
+        set_model(model)
+
+        # TODO:
+        # needs to check if gamma&beta are the same?
+        # Move every BN to eval mode and update the corresponding BN node params in the graph
+        for name, module in model.named_modules():
+            if isinstance(module, torch.nn.BatchNorm2d):
+                module.eval()
+                bn_node = graph.find_node_by_name(name)[0]
+                bn_node_weights = {GAMMA: module.weight.detach().cpu().numpy(),
+                                   BETA: module.bias.detach().cpu().numpy(),
+                                   MOVING_MEAN: module.running_mean.detach().cpu().numpy(),
+                                   MOVING_VARIANCE: module.running_var.detach().cpu().numpy()}
+                # TODO:
+                # check if to set w by key?
+                bn_node.weights = copy.deepcopy(bn_node_weights)
+
+    def quantized_model_builder_for_second_moment_correction(self, graph, fw_info):
+        """
```

Comment:
This method is duplicated for both Keras and PyTorch implementations. Consider moving it to common